db/hints/internal/hint_storage.cc

```diff
@@ -74,16 +73,14 @@ future<> scan_shard_hint_directories(const fs::path& hint_directory,
 ///                     K = <number of shards during the previous boot> * <number of endpoints
 ///                             for which hints where ever created>
 ///
-/// \note Should be called from a seastar::thread context.
-///
 /// \param hint_directory directory to scan
 /// \return a map: ep -> map: shard -> segments (full paths)
-hints_segments_map get_current_hints_segments(const fs::path& hint_directory) {
+future<hints_segments_map> get_current_hints_segments(const fs::path& hint_directory) {
     hints_segments_map current_hint_segments{};
 
     // Shard level.
-    scan_shard_hint_directories(hint_directory,
-            [&current_hint_segments] (fs::path dir, directory_entry de, unsigned shard_id) {
+    co_await scan_shard_hint_directories(hint_directory,
```

Comment:
> db/hints: Move to coroutines in hint_storage.cc 

Nit: "Move to" at the beginning of the commit message strongly suggests that you are moving code from one place to another. We usually use the term "coroutinize" to indicate that some code was adjusted to use coroutines.